### PR TITLE
update kn service resources to show deployment with pod  status

### PR DESCRIPTION
**Pull request by invincibleJai**
update knative service sidebar resources to show deployment with pod status

Tracks: https://jira.coreos.com/browse/ODC-2158

Screenshot 2019-11-15 at 7 07 00 PM

### CODE REVIEW

File: frontend/packages/knative-plugin/src/components/overview/RevisionsOverviewList.tsx [hunk 1]
```diff
@@ -40,6 +45,30 @@ const RevisionsOverviewListItem: React.FC<RevisionsOverviewListItemProps> = ({
...
+            </div>
+            {revision.kind === RevisionModel.kind && (
+              <div className="col-lg-4 col-md-4 col-sm-4">
+                <PodRing
+                  pods={current ? current.pods : []}
+                  obj={revision}
+                  rc={current && current.obj}
+                  resourceKind={RevisionModel}
+                  path="/spec/replicas"
+                />
```
Comment (sahil143) [inline frontend/packages/knative-plugin/src/components/overview/RevisionsOverviewList.tsx]: you can use `PodStatus` component here and pass number pods through `title` prop.

File: frontend/packages/knative-plugin/src/components/overview/RevisionsOverviewList.tsx [hunk 2]
```diff
@@ -19,19 +20,23 @@ export type RevisionsOverviewListItemProps = {
...
-  },
-}) => {
-  const getTraffic = (revision: string) => {
+  } = service;
+  const getTraffic = (name: string) => {
     if (!traffic || !traffic.length) {
       return null;
     }
-    return `${_.get(_.find(traffic, { revisionName: revision }), 'percent', 0)}%`;
+    return `${_.get(_.find(traffic, { revisionName: name }), 'percent', 0)}%`;
```
Comment (sahil143) [inline frontend/packages/knative-plugin/src/components/overview/RevisionsOverviewList.tsx]: The default value here indicates that `0%` traffic is assigned to a revision whereas in this case, it's not assigned any percentage if it's not in `status.traffic`. Maybe return `null` as default value here.

File: frontend/packages/knative-plugin/src/components/overview/RevisionsOverviewList.tsx (hunk 1 above)
Comment (sahil143) [reply frontend/packages/knative-plugin/src/components/overview/RevisionsOverviewList.tsx]: The pod ring doesn't match the designs right now. If you use PodStatus, you can pass innerRadius, outerRadius, size through props to match it with UX designs.

**Comment by karthikjeeyar** [comment]
@invincibleJai looks like it has lint error

**Comment by invincibleJai** [comment]
@sahil143 have made use of `podStatus` however I see issue with `title` if we use both inner and outer radius and this size as in below

Screenshot 2019-11-14 at 5 26 32 PM

current implementation have used inner and outer radius and tooltip to show the status without title as in below

Screenshot 2019-11-14 at 5 37 19 PM

let me know if we can go with second one cc @serenamarie125

**Comment by invincibleJai** [comment]
Thanks @sahil143 have updated the PR with your changes for showing title correctly and `titleComponent` helps. However with tooltip i still see being there as `isVisible` is not working as expected.

File: frontend/packages/console-shared/src/components/pod/PodStatus.tsx [hunk 3]
```diff
@@ -24,6 +24,7 @@ type PodStatusProps = {
   data: ExtPodKind[];
   showTooltip?: boolean;
   title?: string;
+  titleComponent?: React.ReactElement<any>;
```
Comment (vikram-raj) [inline frontend/packages/console-shared/src/components/pod/PodStatus.tsx]: A type is not defined.

```suggestion
  titleComponent?: React.ReactElement;
```

**Comment by serenamarie125** [comment]
this is looking really good, is there any way to make the colored part of the donut a bit thinner?
possibly keep the radius of the outer circle the same, but increasing the inner circle by a pixel?

FYI @sspeiche

**Comment by invincibleJai** [comment]
@serenamarie125 have increased the inner radius by 1 px and updated description image, PTAL

/assign @christianvogt

**Pull request merged**